cohere/compass/utils.py

```diff
@@ -47,7 +48,10 @@ def imap_queued(
                 futures_set, return_when=futures.FIRST_COMPLETED
             )
             for future in done:
-                yield future.result()
+                try:
+                    yield future.result()
+                except Exception as e:
+                    logger.error(f"Error in processing file: {e}")
```

Comment:
Maybe use “exception” instead of “error” to have the stack trace also logged?

Reply:
+1